src/ocean/application/Daemon.d

```diff
@@ -0,0 +1,814 @@
+    import ConfigFiller = ocean.util.config.ConfigFiller;
+    import ocean.util.log.Appender;
+    import ocean.util.log.AppendFile;
+    import LogConfig = ocean.util.log.Config;
+    import ocean.util.log.LayoutDate;
+    import ocean.util.log.Logger;
+    import ocean.util.log.Stats;
+    import ocean.sys.Stats;
+
+    import ConfigOverrides = ocean.application.components.ConfigOverrides;
+    import ocean.application.components.OpenFiles;
+    import ocean.application.components.PidLock;
+    import ocean.application.components.Signals;
+    import TaskScheduler = ocean.application.components.TaskScheduler;
+    import ocean.application.components.Timers;
+    import ocean.application.components.UnixSocketCommands;
+    import Version = ocean.application.components.Version;
+
+    /// Settings passed to the constructor.
+    public struct Settings
+    {
+        /// Application name. Mandatory.
+        istring name;
+
+        /// Application description string.
+        istring desc;
+
+        /// How the program is supposed to be invoked.
+        istring usage;
+
+        /// Long description of what the program does and how to use it.
+        istring help;
+
+        /// Application version info. Mandatory.
+        Version.VersionInfo ver;
+
+        /// Configuration files to parse.
+        istring[] config_files = [ "etc/config.ini" ];
+
+        /// If true, errors when parsing configuration files are treated as
+        /// fatal -- the app will not start.
+        bool config_errors_fatal = true;
+
+        /// Set of signals to handle.
+        int[] signals;
+
+        /// Set of signals to ignore. Delivery of the signals specified in this
+        /// set will have no effect on the application -- they are not passed to
+        /// the default signal handler.
+        int[] ignore_signals;
+
+        /// Default scheduler configuration to be used if no scheduler settings
+        /// are specified in the config file.
+        Scheduler.Configuration scheduler_config;
+    }
+
+    /// Settings passed to the constructor.
+    protected Settings settings;
+
+    /// Command line args.
+    protected Arguments args;
+
+    /// Configuration parser to use to parse the configuration files.
+    protected ConfigParser config;
+
+    /// Stats logger.
+    protected StatsLog stats_log;
+
+    /// Stats logging config. (Also required for timer setup.)
+    private StatsLog.Config stats_log_config;
+
+    /// Set of files opened by the application that should be reopened on signal.
+    protected OpenFiles files;
+
+    /// Unix socket helper.
+    protected UnixSocketCommands unix_socket;
+
+    /// Application-level timers.
+    protected Timers timers;
+
+    /// PID lock helper.
+    private PidLock pid;
+
+    /// Signal handlers.
+    private Signals signals;
+
+    /// Process stats tracker.
+    private CpuMemoryStats process_stats;
+
+    // TODO: global process monitor (see ocean.io.select.client.EpollProcess)
```

Comment:
Still todo?

Reply:
This is more of a "might be a nice thing to add in the future". Not intended to be addressed in this PR. (We could add an issue for this, when this PR is finalised.)